buildx.sh

```diff
@@ -0,0 +1,163 @@
+#!/bin/bash
+
+# Copyright 2020 Adevinta
+set -e
+
+
+function main() {
+  echo "" # see https://github.com/actions/toolkit/issues/168
+
+  # Aapt to be compatible with artifactory registry
+  INPUT_REGISTRY=${INPUT_REGISTRY:-$ARTIFACTORY_DOCKER_REGISTRY}
+  INPUT_USERNAME=${INPUT_USERNAME:-$ARTIFACTORY_USER}
+  INPUT_PASSWORD=${INPUT_PASSWORD:-$ARTIFACTORY_PWD}
+
+  # Adapt to be compatible with travis-ci
+  INPUT_USERNAME=${INPUT_USERNAME:-$DOCKER_USERNAME}
+  INPUT_PASSWORD=${INPUT_PASSWORD:-$DOCKER_PASSWORD}
+  INPUT_NAME=${INPUT_NAME:-$GITHUB_REPOSITORY}
+  INPUT_NAME=${INPUT_NAME:-$TRAVIS_REPO_SLUG}
+  GITHUB_SHA=${GITHUB_SHA:-$TRAVIS_COMMIT}
+  SHORT_SHA=${GITHUB_SHA::7}
+  GITHUB_REF=${GITHUB_REF:-$TRAVIS_PULL_REQUEST_BRANCH}   # In case of pull request we want the originating branch, not the target.
+  GITHUB_REF=${GITHUB_REF:-$TRAVIS_BRANCH}
+  GIT_BRANCH=$(echo "${GITHUB_REF}" | sed -e "s/refs\/heads\///g" | sed -e "s/\//-/g")
+
+  if [ "$GITHUB_REF_TYPE" == "tag" ]; then
+    INPUT_TAG=${GITHUB_REF_NAME}
+  else
+    INPUT_TAG=${TRAVIS_TAG}
+  fi
+
+  echo "INPUT_NAME=${INPUT_NAME}"
+  echo "GITHUB_SHA=${GITHUB_SHA}"
+  echo "GITHUB_REF=${GITHUB_REF}"
+  echo "INPUT_TAG=${INPUT_TAG}"
+  echo "GIT_BRANCH=${GIT_BRANCH}"
+
+  # Prevents building and pushing images when pull_request.
+  # This is an additional control in case we allow in Travis access to secret env variables from
+  # pull requests from forks (DOCKER_PASSWORD).
+  # This could lead to generate "edge" tags if pushed from master branch, or overwritting a branch tag.
+  # TODO: For Github Actions we should review what is the behavoiur.
+  if [ "$TRAVIS_EVENT_TYPE" == "pull_request" ]; then
+    >&2 echo "Disabled build and push on pull-requests"
+    exit 1
+  fi
+
+  sanitize "${INPUT_NAME}" "name"
+  sanitize "${INPUT_USERNAME}" "username"
+  sanitize "${INPUT_PASSWORD}" "password"
+
+  REGISTRY_NO_PROTOCOL=${INPUT_REGISTRY#"https://"}
+  if uses "${INPUT_REGISTRY}" && ! isPartOfTheName "${REGISTRY_NO_PROTOCOL}"; then
```

Comment:
I maybe not getting something here, but the function `uses` is not defined in this script, right?

Reply:
You are absolutelly right,fixed. Thanks!!